govcd/adminvdc_test.go

```diff
@@ -0,0 +1,239 @@
+						Limit:     1024,
+					},
+					Memory: &types.CapacityWithUsage{
+						Allocated: 1024,
+						Limit:     1024,
+					},
+				},
+			},
+			VdcStorageProfile: []*types.VdcStorageProfile{&types.VdcStorageProfile{
+				Enabled: true,
+				Units:   "MB",
+				Limit:   1024,
+				Default: true,
+				ProviderVdcStorageProfile: &types.Reference{
+					HREF: providerVdcStorageProfileHref,
+				},
+			},
+			},
+			NetworkPoolReference: &types.Reference{
+				HREF: networkPoolHref,
+			},
+			ProviderVdcReference: &types.Reference{
+				HREF: providerVdcHref,
+			},
+			IsEnabled:            true,
+			IsThinProvision:      true,
+			UsesFastProvisioning: true,
+		}
+
+		if allocationModel == "Flex" {
+			vdcConfiguration.IsElastic = &trueValue
+			vdcConfiguration.IncludeMemoryOverhead = &trueValue
+		}
+
+		vdc, _ := adminOrg.GetVDCByName(vdcConfiguration.Name, false)
+		if vdc != nil {
+			err = vdc.DeleteWait(true, true)
+			check.Assert(err, IsNil)
+		}
+
+		// expecting fail due missing value
+		task, err := adminOrg.CreateOrgVdcAsync(vdcConfiguration)
+		check.Assert(err, Not(IsNil))
+		check.Assert(task, Equals, Task{})
+		// From vCD 10+ error changed
+		if vcd.client.Client.APIVCDMaxVersionIs("> 32.0") {
+			check.Assert(err.Error(), Equals, "error retrieving vdc: API Error: 400:  is not a valid unit. Please use MB")
+		} else {
+			check.Assert(err.Error(), Equals, "VdcConfiguration missing required field: ComputeCapacity[0].Memory.Units")
+		}
+		vdcConfiguration.ComputeCapacity[0].Memory.Units = "MB"
+
+		vdc, err = adminOrg.CreateOrgVdc(vdcConfiguration)
+		check.Assert(vdc, NotNil)
+		check.Assert(err, IsNil)
+
+		AddToCleanupList(vdcConfiguration.Name, "vdc", vcd.org.Org.Name, "Test_CreateVdcWithFlex")
+
+		vdc, err = adminOrg.GetVDCByName(vdcConfiguration.Name, true)
+		check.Assert(err, IsNil)
+		check.Assert(vdc, NotNil)
+		check.Assert(vdc.Vdc.Name, Equals, vdcConfiguration.Name)
+		check.Assert(vdc.Vdc.IsEnabled, Equals, vdcConfiguration.IsEnabled)
+		check.Assert(vdc.Vdc.AllocationModel, Equals, vdcConfiguration.AllocationModel)
+
+		err = vdc.DeleteWait(true, true)
+		check.Assert(err, IsNil)
+
+		vdc, err = adminOrg.GetVDCByName(vdcConfiguration.Name, true)
+		check.Assert(err, NotNil)
+		check.Assert(vdc, IsNil)
+	}
+}
+
+// Tests VDC by updating it and then asserting if the
+// variable is updated.
+func (vcd *TestVCD) Test_UpdateVdcFlex(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+
+	// Flex vCD supported from 9.7 vCD
+	if vcd.client.Client.APIVCDMaxVersionIs("< 32.0") {
+		check.Skip(fmt.Sprintf("Test %s requires vCD 9.7 or higher", check.TestName()))
+	}
+
+	adminOrg, vdcConfiguration, err := setupVDc(vcd, check, "Flex")
+	check.Assert(err, IsNil)
+
+	// Refresh so the new VDC shows up in the org's list
+	err = adminOrg.Refresh()
+	check.Assert(err, IsNil)
+
+	adminVdc, err := adminOrg.GetAdminVDCByName(vdcConfiguration.Name, false)
```

Comment:
```suggestion
	adminVdc, err := adminOrg.GetAdminVDCByName(vdcConfiguration.Name, true)
```
Using this construct will make the explicit refresh above unnecessary

Reply:
changed